runtime/src/main/scala/akka/grpc/GrpcServiceException.scala

```diff
@@ -50,7 +54,12 @@ object GrpcServiceException {
   }
 
   def apply(ex: StatusRuntimeException): GrpcServiceException = {
-    new GrpcServiceException(ex.getStatus, new RichGrpcMetadataImpl(ex.getStatus, ex.getTrailers))
+    new GrpcServiceException(
+      ex.getStatus,
+      new RichGrpcMetadataImpl(
+        ex.getStatus,
```

Comment:
Unrelated to this PR, but why we use `RichGrpcMetadataImpl`, but above we just use `GrpcMetadataImpl`?

Reply:
We want to propagate the structured error details if present, but maybe cleaner and more correct to select between GrpcMetadataImpl and RichGrpcMedataImpl on the presence of trailers instead of giving an empty metadata like I did. I'll do that instead.

Reply:
Hmm, looking at it some more, actually, I'm not so sure why we are not passing `RichGrpcMetadataImpl` above, that looks like the right thing to do.

Reply:
[312a11a](https://github.com/akka/akka-grpc/pull/1855/commits/312a11a0dc1b08d0e2657a9210df0dd3e76773a5) should do the trick